docs/user/exercises/programming-repository-access.inc

```diff
@@ -0,0 +1,130 @@
+  - **Teaching Assistant (TA) Assignment**: An assignment repository created by a teaching assistant for themself.
+  - **Instructor Assignment**: An assignment repository created by an editor or instructor for themselves. Not available for exam exercises.
+  - **Student Practice**: A student's practice repository copied either from the template repository, or from their assignment repository. Can only be created after the due date of the exercise has passed. Not available for exam exercises.
+  - **Teaching Assistant (TA) Practice**: A practice repository created by a teaching assistant for themself. Not available for exam exercises.
+  - **Instructor Practice**: A practice repository created by an editor or instructor for themselves. Not available for exam exercises.
+  - **Instructor Exam Test Run**: A test run repository created by an editor or instructor for an exam. An instructor can create an exam test run before the start date of the exam to allow the instructor to test the exam from a student perspective before releasing it. This repository should be deleted before the exam is conducted.
+
+
+The different roles are:
+
+  - **Student (S)**: A student in the course.
+  - **Teaching Assistant (TA)**: A teaching assistant (tutor) in the course.
+  - **Editor**: An editor in the course.
+  - **Instructor (I)**: An instructor in the course.
+
+Note: Editors and instructors are included in the role "Instructor" in the table as both roles have the same access rights.
+
+The different points in time are:
+
+  - **Before start**: Before the exercise start date for a course exercise, before the exam start date for an exam exercise.
+  - **Working time**: After the exercise start date for a course exercise, after the exam release date for an exam exercise, before the due date for a course exercise, before the exam end date for an exam exercise.
+  - **After due**: After the due date for a course exercise, after the exam end date for an exam exercise.
+
+Note: For the Instructor Exam Test Run Repository, the point in time "Before start" is the start date of the test run, and the point in time "After due" is the end date of the test run. Both are before the exam start date.
+
+Read access **(R)** includes `git fetch`, `git clone`, and `git pull`, if you are using your local Git client to access the repository.
+Write access **(W)** corresponds to `git push` if you are using your local Git client to access the repository.
+
++--------------------------+------+---------------+--------+
+| Repository type          | Role | Point in time | Access |
++==========================+======+===============+========+
+| Base                     | S    | all           | none   |
++--------------------------+------+---------------+--------+
+| Base                     | TA   | all           | R      |
++--------------------------+------+---------------+--------+
+| Base                     | I    | all           | R/W    |
```

Comment:
```suggestion
| Base                     | S    | all           | none   |
|                          +------+---------------+--------+
|                          | TA   | all           | R      |
|                          +------+---------------+--------+
|                          | I    | all           | R/W    |
```
Rows can be joined by omitting the separator. This could reduce the repetitions and make the table structure clearer.

![grafik](https://user-images.githubusercontent.com/64250573/230740773-2bde0bb4-56c2-4c6b-868f-96510328d35c.png)